src/core/qgsstoredexpressionmanager.cpp

```diff
@@ -0,0 +1,147 @@
+/***************************************************************************
+                          qgsstoredexpressionmanager.cpp
+                             -------------------
+    begin                : August 2019
+    copyright            : (C) 2019 David Signer
+    email                : david at opengis dot ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsstoredexpressionmanager.h"
+#include "qgis.h"
+
+
+#include <QDomElement>
+
+QString QgsStoredExpressionManager::addStoredExpression( const QString &name, const QString &expression, const QgsStoredExpression::Category &tag )
+{
+  Q_UNUSED( tag );
+
+  QgsStoredExpression storedExpression( name, expression, tag );
+
+  mStoredExpressions.append( storedExpression );
+
+  return storedExpression.id;
+}
+
+void QgsStoredExpressionManager::removeStoredExpression( const QString &id )
+{
+  int i = 0;
+  for ( const QgsStoredExpression &storedExpression : qgis::as_const( mStoredExpressions ) )
```

Comment:
A nicer approach here would be:

    mStoredExpressions.erase( std::remove_if( mStoredExpressions.begin(), mStoredExpressions.end(), [ & ]( const QgsStoredExpression & storedExpression )
    {
      return storedExpression.id == id;
    } ), mStoredExpressions.end() );

Reply:
Ok, nice.

Reply:
I keep the original implementation, so it's same like in the other methods and easier to read.